lib/src/query.c

```diff
@@ -3103,28 +3103,34 @@ void ts_query_cursor_exec_with_options(
   }
 }
 
-void ts_query_cursor_set_byte_range(
+bool ts_query_cursor_set_byte_range(
   TSQueryCursor *self,
   uint32_t start_byte,
   uint32_t end_byte
 ) {
   if (end_byte == 0) {
     end_byte = UINT32_MAX;
+  } else if (start_byte > end_byte) {
+    return false;
   }
   self->start_byte = start_byte;
   self->end_byte = end_byte;
+  return true;
 }
 
-void ts_query_cursor_set_point_range(
+bool ts_query_cursor_set_point_range(
   TSQueryCursor *self,
   TSPoint start_point,
   TSPoint end_point
 ) {
   if (end_point.row == 0 && end_point.column == 0) {
     end_point = POINT_MAX;
+  } else if (point_gt(start_point, end_point)) {
+    return false;
   }
```

Comment:
remove the `else`, since at this point we've set the end point to `POINT_MAX`, so the `gt` comparison will evaluate to false
```suggestion
  if (point_gt(start_point, end_point)) {
    return false;
  }
```